app/src/main/java/com/oracle/javafx/scenebuilder/app/DocumentWindowController.java

```diff
@@ -1573,7 +1573,11 @@ private void performPaste() {
         } else {
             assert isTextInputControlEditing(focusOwner);
             final TextInputControl tic = getTextInputControl(focusOwner);
-            tic.paste();
+            // TODO: Once the issue https://bugs.openjdk.java.net/browse/JDK-8205915
```

Comment:
Change to https://bugs.openjdk.java.net/browse/JDK-8280057

Reply:
The TODO is now changed to 8280057. How about the switch? Shall I add a failing test for versions > 18? How to go forward here? Once a while, I'll pull OpenJFX to learn but as of now, I'd like to stay with Scene Builder, there's enough to learn at this very moment.

Reply:
The issue JDK-8280057 has been closed as duplicated of https://bugs.openjdk.java.net/browse/JDK-8088897 and there might be a fix "soon" for it with https://github.com/openjdk/jfx/pull/694, which is possibly targeting JavaFX 19.

For now I guess there is no other way than apply your patch, and simply keep an eye on the fix on OpenJFX. Once it is integrated and released, we can decide if we need to apply a condition based on a given version (i.e. 18.0.1, 19,  or 19-ea+3).

Adding a unit test might be hard. A system/manual test might work, but we don't have them in Scene Builder.

Reply:
We could create a Scene Builder 19 milestone and assign an issue to it to remove the fix as soon as OpenJFX has received an appropriate fix. @AlmasB What do you think about it? This would not require any further code and we could track it.

Reply:
Just quickly reading between the lines here.. If I understood correctly, this fix is temporary and will break when openjfx gets a proper fix?

If we are to merge, we do need to keep track of this and possibly keep delaying from milestone to another until the fix is there -- this sounds like a high cost for maintenance.

Could we not have a preference setting that allows enabling / disabling this at runtime? Then, we can future-proof this and fix appropriately without users having to wait for one.

Reply:
Yes it will break, when JavaFX is fixed.

Would you expose this setting in Preferences dialog?

An alternate option would be to check for the JavaFX version used and disable this behavior when using any JavaFX version >= 19/19ea.

Reply:
If you like @AlmasB, you can assign an issue to me so I'll keep an eye on this item.

Reply:
We could add a 2nd preferences page (Platform specific settings) in the existing preferences window.
There we could offer a toggle such as: `Alternate Text Input Control paste behavior = [x]`. How about that?

Reply:
Hi @AlmasB, there is now a preference in the Preferences dialog where the behavior can be turned off. It is enabled by default on MacOS. Also the toggle is disabled for other operating systems. A tool tip provides short description.

<img width="615" alt="Bildschirmfoto 2022-01-21 um 17 47 32" src="https://user-images.githubusercontent.com/22102800/150569265-d45f20cc-4e97-4abb-8c03-ab986ccfdae7.png">